### DISCUSSION TITLE
Testing http service on ec2

### DISCUSSION THREAD

**Discussion by c0debreaker** [gruntwork-io/knowledge-base#726] [How do I]
I just learned about terratest few days ago. I'd like to use it as my integration test in gitlab-ci specifically before the real deployment of the application on an ec2. How do I use terratest to check a web service running on port 8080? There is a /health endpoint which returns "Nginx is up".

---

<ins datetime="2023-06-06T05:09:56Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110229">Tracked in ticket #110229</a></p>
</ins>

**Comment by denis256** [comment]
Hello,
can be used functions from `http_helper` package like `HTTPDoWithValidation() / HTTPDoWithValidationE()`  

Examples:

https://github.com/gruntwork-io/terratest/blob/master/modules/http-helper/http_helper_test.go#L47
https://github.com/gruntwork-io/terratest/blob/master/modules/k8s/service_test.go#L93
https://github.com/gruntwork-io/terratest/blob/master/test/terraform_http_example_test.go#L16

**Reply by c0debreaker** [reply]
Thank you so much! I'll check out the links now.
